File: leetcode/167_two_sum_II/main.go
package main

import "fmt"

func main()  {
	fmt.Println(twoSum([]int{1, 2, 3}, 3))
	fmt.Println(twoSum([]int{0, 2, 3}, 3))
	fmt.Println(twoSum([]int{-1, 0, 1}, 1))
	fmt.Println(twoSum([]int{0, 1}, 1))
}

func twoSum(nums []int, target int) []int {
 	left := 0
 	right := len(nums) - 1 

	for {
		if nums[left] + nums[right] < target {
			left += 1
		} else if nums[left] + nums[right] > target {
			right -= 1
		} else {
			return []int{left+1, right+1}
		}
	}
}

File: leetcode/167_two_sum_II/main.rs
fn main() {
    println!("{:?}", two_sum(vec![1, 2], 3));
    println!("{:?}", two_sum(vec![-11, 2, 11], 0));
    println!("{:?}", two_sum(vec![-11, 2, 11, 22], 0));
}

fn two_sum(nums: Vec<i32>, target: i32) -> Vec<usize> {
    let mut left = 0;
    let mut right = nums.len() - 1;
    
    loop {
        if nums[left] + nums[right] == target {
            return vec![left+1, right+1];
        } else if nums[left] + nums[right] > target {
            right -= 1;
        } else {
            left += 1;
        }
    }

}